chaste_codegen/rush_larsen_opt_model.py

```diff
@@ -13,7 +15,25 @@ def _get_extended_equations_for_ionic_vars(self):
         return partial_eval(super()._get_extended_equations_for_ionic_vars(),
                             [eq.lhs for eq in self._equations_for_ionic_vars])
 
-    def _format_alpha_beta_eqs(self):
+    def _update_state_vars(self):
+        """Updates formatting of state vars to make sure the correct ones are included in the output"""
         self._vars_for_template['derivative_alpha_beta_eqs'] = \
-            partial_eval(self._vars_for_template['derivative_alpha_beta_eqs'], self._vars_in_derivative_alpha_beta)
-        return super()._format_alpha_beta_eqs()
+            partial_eval(self._vars_for_template['derivative_alpha_beta_eqs'], self._vars_in_derivative_alpha_beta,
+                         keep_multiple_usages=False)
+        # Get all used variables for derivative eqs
+        deriv_variables = set()
+        for eq in self._vars_for_template['derivative_alpha_beta_eqs']:
+            deriv_variables.update(eq.rhs.free_symbols)
+
+        for sv in self._formatted_state_vars:
+            sv['in_ab'] = sv['sympy_var'] in deriv_variables
+
+    def _format_alpha_beta_eqs(self):
```

Comment:
Is there a reason we can't use the base class method here? Why the different logic for `in_eqs_excl_voltage` (which seems to be the only difference)?

Reply:
I see where you're coming from. 
The thing is that for the self._derivative_eqs_excl_voltage is a subset of the original derivative equations, so not partially evaluated. When you apply the partial eval,the only equation that's not in there is: dx/dt = ... so instead of applying the partial eval I just wrote that. (The rest of the result of partial eval you won't see anyway in these types of models).

 Maybe it would be more logical though to just apply the partial eval?

Reply:
If you get the same result either way, I'd be inclined to avoid the code duplication I think.

Reply:
Or at least add a comment to explain.

Reply:
Ok makse sense I rewrote it to apply partial eval and use super class formatting rules.